commands/social/store/store.js

```diff
@@ -106,26 +70,109 @@ class Store extends Social {
-        if (items.length === 0) return message.response(undefined, "Baka... nothing is for sale!");
-        message.channel.send(`= ${message.guild.name} General Store =\n` + items.map(item => 
-          `${message.guild.roles.get(item.id.toString()).name}${" ".repeat(40 - message.guild.roles.get(item.id.toString()).name.length)}::  ${parseInt(item.price) === 0 ? "FREE" : `₲${parseInt(item.price).toLocaleString()}`} :: ${message.member.roles.has(item.id) ? "✓" : ""}`).join("\n"), { code: "asciidoc" }
+        if (!items.length) return  message.channel.send(`= ${message.guild.name} Store =\n` + "\n" + "= Tokens = \n" + `Slots${" ".repeat(20 - "Slots".length)}:: ₲${message.settings.tokenPrice} :: ${message.member.inventory["tokens"]} \n`, { code: "asciidoc" });
+        message.channel.send(`= ${message.guild.name} Store =\n` + "\n" + "= Tokens = \n" + `Slots${" ".repeat(20 - "Slots".length)}:: ₲${message.settings.tokenPrice} :: ${message.member.inventory["tokens"]} \n \n`+ "= Roles For Sale = \n"+ items.map(item => 
+          `${message.guild.roles.get(item.id.toString()).name}${" ".repeat(20 - message.guild.roles.get(item.id.toString()).name.length)}::  ${parseInt(item.price) === 0 ? "FREE" : `₲${parseInt(item.price).toLocaleString()}`} ${message.member.roles.has(item.id) ? ":: ✓" : ""}`).join("\n"), { code: "asciidoc" }
         );
       }
     }
   }
+  async buyToken(name, message, client) {
+    if (parseInt(message.settings.tokenPrice) > parseInt(message.member.score.points)) {
+      return message.channel.send(`You currently have ₲${parseInt(message.member.score.points).toLocaleString()}, but the token costs ${parseInt(message.settings.tokenPrice).toLocaleString()}!`);
+    }
+    const filter = m => m.author.id === message.author.id;
+    const response = await client.awaitReply(message, `Are you sure you want to purchase a Slot Token for ₲${message.settings.tokenPrice}?`, filter, undefined, null);
+    if (["y", "yes"].includes(response.toLowerCase())) {
+    
+      message.member.takePoints(parseInt(message.settings.tokenPrice));
+      await message.member.giveItem("tokens", 1);
+      message.channel.send("You have bought a token");
+    
+    } else
+    
+    if (["n", "no", "cancel"].includes(response.toLowerCase())) {
+      message.response(undefined, "Transaction cancelled.");
+    }
+  }
+  async buyRole(name, message, client) {
+    const item = client.store.filter(i => i.name.toLowerCase().includes(name.toLowerCase()));
+    
+    if (item.size > 1) return message.reply(`B..Baka! Be more specific more than that, there is more than one item on sale with ${name} as their name`);
+    if (!item) return message.channel.send("That item doesn't exist, Please make sure it is spelled correctly");
+    if (message.member.roles.has(item.array()[0].id)) return message.channel.send("You already have the role :facepalm: ");
+    
+    if (parseInt(item.array()[0].price) > parseInt(message.member.score.points)) {
```

Comment:
transform that collection to an array is pointless here and uses more time and resources aswell as what i mentioned in line 95 again